cinema/views.py

```diff
@@ -100,9 +102,26 @@ def get_serializer_class(self):
 
         if self.action == "retrieve":
             return MovieDetailSerializer
+        if self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        url_path="upload-image",
+        permission_classes=[IsAuthenticated]
```

Comment:
You have defined IsAdminOrIfAuthenticatedReadOnly permission class, it is working properly in this case.